src/Microsoft.Health.Fhir.CosmosDb.Initialization/Features/Storage/DataPlaneCollectionSetup.cs

```diff
@@ -0,0 +1,149 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Abstractions.Exceptions;
+using Microsoft.Health.Core.Features.Context;
+using Microsoft.Health.Fhir.CosmosDb.Core;
+using Microsoft.Health.Fhir.CosmosDb.Core.Configs;
+using Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.Versioning;
+using Polly;
+using Polly.Retry;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Initialization.Features.Storage
+{
+    public class DataPlaneCollectionSetup : ICollectionSetup
+    {
+        private readonly ILogger<DataPlaneCollectionSetup> _logger;
+        private const int CollectionSettingsVersion = 3;
+
+        public DataPlaneCollectionSetup(ILogger<DataPlaneCollectionSetup> logger)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+            _logger = logger;
+        }
+
+        public async Task CreateDatabaseAsync(CosmosClient client, CosmosDataStoreConfiguration cosmosDataStoreConfiguration, AsyncPolicy retryPolicy, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(client, nameof(client));
+            EnsureArg.IsNotNull(cosmosDataStoreConfiguration, nameof(cosmosDataStoreConfiguration));
```

Comment:
We should also add a validation to check if the retry policy is not null.

Reply:
done